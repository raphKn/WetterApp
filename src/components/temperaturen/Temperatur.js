import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import CustomText from '../common/CustomText';

function createStyle(isMinTemp, isMaxTemp) {
    let color;
    if (isMinTemp) {
        color = 'blue';
    } else if (isMaxTemp) {
        color = 'red';
    }

    return StyleSheet.create({
        text: {
            fontSize: 32,
            color,
        },
    });
}

const styles = StyleSheet.create({
    container: {
        marginHorizontal: 6,
    },
    underline: {
        alignSelf: 'center',
    },
});

function createUnderline(isMinTemp, isMaxTemp) {
    if (isMinTemp) {
        return 'min.';
    }
    if (isMaxTemp) {
        return 'max.';
    }
    return 'aktuell';
}

const Temperatur = props => (<View style={styles.container}>
    <CustomText style={createStyle(props.isMinTemp, props.isMaxTemp).text}>{props.value}</CustomText>
    <CustomText style={styles.underline}>{createUnderline(props.isMinTemp, props.isMaxTemp)}</CustomText>
</View>);

Temperatur.propTypes = {
    value: PropTypes.string.isRequired,
    isMinTemp: PropTypes.bool,
    isMaxTemp: PropTypes.bool,
};

Temperatur.defaultProps = {
    isMinTemp: false,
    isMaxTemp: false,
};

export default Temperatur;