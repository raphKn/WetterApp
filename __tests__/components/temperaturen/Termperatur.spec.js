import React from 'react';
import renderer from 'react-test-renderer';
import Temperatur from '../../../src/components/temperaturen/Temperatur';

describe('Temperatur', () => {
    it('renders maxTemp correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <Temperatur value="12" isMaxTemp />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders normal Temp correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <Temperatur value="12" />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders min Temp correctly in a snapshot', () => {
        const wrapper = renderer.create(
            <Temperatur value="12" isMinTemp />);
        expect(wrapper).toMatchSnapshot();
    });
});
