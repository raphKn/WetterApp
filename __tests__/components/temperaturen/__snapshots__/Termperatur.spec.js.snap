// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperatur renders maxTemp correctly in a snapshot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 6,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#212121",
        },
        Object {
          "color": "red",
          "fontSize": 32,
        },
      ]
    }
  >
    12
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#212121",
        },
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    max.
  </Text>
</View>
`;

exports[`Temperatur renders min Temp correctly in a snapshot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 6,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#212121",
        },
        Object {
          "color": "blue",
          "fontSize": 32,
        },
      ]
    }
  >
    12
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#212121",
        },
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    min.
  </Text>
</View>
`;

exports[`Temperatur renders normal Temp correctly in a snapshot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 6,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#212121",
        },
        Object {
          "color": undefined,
          "fontSize": 32,
        },
      ]
    }
  >
    12
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#212121",
        },
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    aktuell
  </Text>
</View>
`;
